ledfx/utils.py

```
@@ -1463,6 +1463,25 @@ def clip_at_limit(numbers, limit):
+        if parsed.scheme and parsed.netloc
+        else ""
+    )
+    headers = {
+        "User-Agent": (
+            "Mozilla/5.0 (Windows NT 10.0; Win64; x64) "
+            "AppleWebKit/537.36 (KHTML, like Gecko) "
+            "Chrome/141.0.0.0 Safari/537.36"
+        ),
+        "Referer": origin,  # helps with sites that block direct hotlinks (e.g., JSTOR)
+    }
+    return urllib.request.Request(url, headers=headers)
```

Ruff (0.13.3)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)